erc20-load-test-tool/src/main/java/com/vmware/ethereum/service/WorkloadCommand.java

```diff
@@ -30,53 +30,86 @@
 import static java.time.Duration.between;
 import static java.time.Instant.now;
 
+import com.vmware.ethereum.config.Web3jConfig;
+import java.io.IOException;
 import java.time.Duration;
 import java.time.Instant;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CountDownLatch;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.BatchRequest;
+import org.web3j.protocol.core.BatchResponse;
 import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.protocol.exceptions.TransactionException;
 import org.web3j.tx.response.EmptyTransactionReceipt;
+import org.web3j.tx.response.TransactionReceiptProcessor;
 
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class WorkloadCommand implements Runnable {
 
+  private final MetricsService metrics;
+  private final Web3j web3j;
+  private final Web3jConfig web3jConfig;
+  private final TransactionReceiptProcessor transactionReceiptProcessor;
   private final SecureTokenApi api;
   private final CountDownLatch countDownLatch;
-  private final MetricsService metrics;
+  private BatchRequest batchRequest = null;
```

Comment:
An instance of BatchRequest should be created per request. Sharing it might cause race condition.

Reply:
We need to share this batchRequest among the requests, as according to the arrival rate tx are created which needs to be added to the batchRequest. I couldn't get it work otherwise